### PR TITLE
feat: add optional trace to CairoVM struct and add enable trace flag

### LINKED ISSUE #9: feat: add optional trace to CairoVM struct and add enable trace flag
Right now the CairoVM is [missing an optional trace field](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/vm/core.zig#L14), which when we are in `enable-trace` mode should be updated with every step of the virtual machine. The trace is a vector of `TraceEntry` of the registers `pc`, `ap`, and `fp`. In `runInstruction` `[function](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/vm/core.zig#L104)` we need to push a new `TraceEntry` struct that receives the registers from `RunContext` [register holder](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/vm/run_context.zig#L12). 

It's OK to push a default value of `RunContext` for the moment but when [this](https://github.com/keep-starknet-strange/cairo-zig/issues/10) is complete the actual value needs to be pushed. 

Add the `enable-trace` flag in [cmd](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/cmd/cmd.zig).

### PR DESCRIPTION
**Pull request by itzlambda**
Fixes: #9 

I am new to zig so let me know if something is non idiomatic or there is a better way to do it.

### CODE REVIEW

File: src/vm/core.zig [hunk 1]
```diff
@@ -102,6 +109,9 @@ pub const CairoVM = struct {
     // # Arguments
     /// - `instruction`: The instruction to run.
     pub fn runInstruction(self: *CairoVM, instruction: *const instructions.Instruction) !void {
+        if (self.trace != null) {
+            try self.trace.?.append(TraceEntry{ .pc = self.run_context.pc, .ap = self.run_context.ap, .fp = self.run_context.fp });
+        }
```
Comment (abdelhamidbakhta) [inline src/vm/core.zig:114]:
```suggestion
if(self.trace) |tracer| {
    try tracer.append(TraceEntry{ .pc = self.run_context.pc, .ap = self.run_context.ap, .fp = self.run_context.fp });
}
```

File: src/vm/core.zig [hunk 2]
```diff
@@ -781,3 +797,84 @@ test "update fp dst felt" {
...
+    // ************************************************************
+    // Check that trace was initialized
+    const trace = vm.trace;
+
+    if (trace != null) {
+        return error.TraceShouldHaveBeenDisabled;
+    }
+}
+
+// TODO: Find a valid Instruction so we can call `runInstruction` to check if traces
```
Comment (abdelhamidbakhta) [inline src/vm/core.zig:855]: remove commented test and create an issue with the content of this test instead please

**Review by AbdelStark — CHANGES_REQUESTED** [review]
left some comments

**Review by AbdelStark — COMMENTED** [review]
I am wondering if we should just make the `config` struct of the cmd in a `config.zig` file and share it between the cli and the core vm. 
Instead of passing individual flags like we do in this PR.
I am worried that it will be harder to maintain if each time we add a new flag in the init function and a new field in the VM.
For example we see here that you had to change each individual test to add the new parameter.
If it was the struct and if the struct has a default function you would not have to change the tests most of the time, only when you need to change the config for an actual test.

**Comment by lambda-0x** [comment]
reopened: #27

**Pull request closed without merging**